agent/logs.go

```diff
@@ -6,15 +6,18 @@ import (
 	"time"
 
 	"github.com/google/uuid"
-	"golang.org/x/exp/slices"
 	"golang.org/x/xerrors"
 
 	"cdr.dev/slog"
 	"github.com/coder/coder/v2/agent/proto"
 	"github.com/coder/coder/v2/codersdk/agentsdk"
 )
 
-const flushInterval = time.Second
+const (
+	flushInterval     = time.Second
+	logOutputMaxBytes = 1 << 20 // 1MiB
+	overheadPerLog    = 21      // found by testing
```

Comment:
We should probably also have an upper bound of buffered logs since these will take up memory, or alternatively, store logs on disk until they can be sent.

If something goes wrong with a command, there's really no telling how much it will spit out, it could be KiB's or it could be GiB's (in extreme cases). We also have limits in place in the API that limit the output to 1 MiB, although I can't recall the details if we store both the head and tail end of logs or simply cutoff at 1 MiB.

Reply:
Not a bad idea.  Can address in a follow on PR.